### ISSUE TITLE
Just-opened narrow sometimes flickers "no messages" before fetch has started

### ISSUE THREAD

**Issue by chrisbobbe** [zulip/zulip-mobile#5152]
```js
const isFetching = fetching.older || fetching.newer || loading;
// [...]
const sayNoMessages = messages.length === 0 && !isFetching;
```

We fudge `fetching.older` and `fetching.newer` as `false` in `fetchingReducer` when this is a narrow we've just opened; one that `state.fetching` doesn't have info for yet.

`fetchMessagesInNarrow` correctly sets `fetching.older` and `fetching.newer` to `true` when it's called. But by the time it's called, it's too late: the `false` values have already made `sayNoMessages` `true`, wrongly, for at least a frame.

Right?

**Comment by chrisbobbe** [comment]
> We fudge `fetching.older` and `fetching.newer` as `false` in `fetchingReducer` when this is a narrow we've just opened; one that `state.fetching` doesn't have info for yet.

Should we be fudging it as `true`, or not fudging it at all?

**Comment by chrisbobbe** [comment]
A user report with screenshots from #5636 (which I'm about to close as a duplicate; please let me know if you think this is wrong):

> In every chat page, the "No messages" text displays for a second before showing a loading placeholder.
> 
> For a better experience, the chat page should start with a loading placeholder and then continue to show no messages or all chats.
> 
> "No messages" text before placeholder
> <img src="https://user-images.githubusercontent.com/90995338/211165969-4efc3378-8f7c-4f4d-9c47-decf379b9d5d.jpg" width="300" alt="No messages"> <img src="https://user-images.githubusercontent.com/90995338/211165965-709ecf6e-9acb-448b-986a-3908e2a5626c.jpg" width="300" alt="Loading placeholder">

**Comment by anshuman-8** [comment]
Yes @chrisbobbe , this seems to be the same issue as #5636 , where `sayNoMessage` gets set as `true` before `fetchMessagesInNarrow` in `chatScreen` causing "No messages" to come up before loading placeholder.
May I work on this issue?

**Comment by anshuman-8** [comment]
Hey @chrisbobbe ,

I have submitted a [pull request](https://github.com/zulip/zulip-mobile/pull/5639) to fix this problem.
Please review it and let me know if anything further is needed.

Thank you.
